packages/angular_devkit/build_angular/test/browser/rebuild_spec_large.ts

```diff
@@ -214,99 +230,130 @@ describe('Browser Builder rebuilds', () => {
-
-          case 4:
-            expect(buildEvent.success).toBe(false);
-            // Restore the file to a error-less state.
-            host.writeMultipleFiles({ 'src/app/app.component.ts': origContent });
-            break;
-
-          case 5:
-            // The fifth build should have everything fixed..
-            expect(buildEvent.success).toBe(true);
-            expect(logger.includes(staticAnalysisError)).toBe(true);
-            break;
-        }
-      }),
-      take(5),
-    ).toPromise();
+    await run.output
+      .pipe(
+        debounceTime(1000),
+        tap(buildEvent => {
+          buildNumber += 1;
+          switch (buildNumber) {
+            case 1:
+              // The first build should error out with a static analysis error.
+              expect(buildEvent.success).toBe(false);
+              expect(logger.includes(staticAnalysisError)).toBe(true);
+              logger.clear();
+              // Fix the static analysis error.
+              host.writeMultipleFiles({ 'src/app/app.component.ts': origContent });
+              break;
+
+            case 2:
+              expect(buildEvent.success).toBe(true);
+              expect(logger.includes(staticAnalysisError)).toBe(false);
+              logger.clear();
+              // Add an syntax error to a non-main file.
+              host.appendToFile('src/app/app.component.ts', `]]]`);
+              break;
+
+            case 3:
+              // The third build should have TS syntax error.
+              expect(buildEvent.success).toBe(false);
+              expect(logger.includes(syntaxError)).toBe(true);
+              expect(logger.includes(staticAnalysisError)).toBe(false);
+              logger.clear();
+              // Fix the syntax error, but add the static analysis error again.
+              host.writeMultipleFiles({
+                'src/app/app.component.ts': origContent.replace(
+                  `'app-root'`,
+                  `(() => 'app-root')()`,
+                ),
+              });
+              break;
+
+            case 4:
+              expect(buildEvent.success).toBe(false);
+              expect(logger.includes(syntaxError)).toBe(false);
+              expect(logger.includes(staticAnalysisError)).toBe(true);
+              logger.clear();
+              // Restore the file to a error-less state.
+              host.writeMultipleFiles({ 'src/app/app.component.ts': origContent });
+              break;
+
+            case 5:
+              // The fifth build should have everything fixed..
+              expect(buildEvent.success).toBe(true);
+              expect(logger.includes(syntaxError)).toBe(false);
+              expect(logger.includes(staticAnalysisError)).toBe(false);
+              break;
+          }
+        }),
+        take(5),
+      )
+      .toPromise();
     await run.stop();
   });
 
   it('rebuilds AOT factories', async () => {
+    if (ivyEnabled) {
+      pending('Does not apply to Ivy.');
```

Comment:
Although the factory files don't exist in Ivy, we still want to ensure that template changes are detected and bundles are updated. I think this test should mostly work as is because it checks changes go into `./dist/main.js`. 

Do you see it broken? I know this was broken because of https://github.com/angular/angular/issues/30947 but I think we have fixes in already. Maybe we need to update the FW version to include https://github.com/angular/angular/pull/30954 that matches https://github.com/angular/angular-cli/pull/14753.

Reply:
It's broken.  I updated the comment to reflect.  I also tried with FW rc.0 and it still failed.

Reply:
cc @petebacondarwin looks like the rebuilds with template changes are still broken.

Reply:
I'll do some more followup testing as well.

Reply:
This test was re-enabled in https://github.com/angular/angular-cli/pull/15013.